client/my-sites/email/form/mailboxes/validators.ts

```diff
@@ -242,14 +281,73 @@ class PasswordValidator extends BaseValidator< string > {
 				regexp.test( character )
 			);
 
-			field.error =
-				PasswordValidator.getPasswordContainsForbiddenCharacterError( firstForbiddenCharacter );
-			return;
+			this.addError(
+				field,
+				PasswordValidator.getPasswordContainsForbiddenCharacterError( firstForbiddenCharacter )
+			);
 		}
 
 		if ( value.endsWith( ' ' ) ) {
-			field.error = PasswordValidator.getPasswordEndsWithSpaceError();
+			this.addError( field, PasswordValidator.getPasswordEndsWithSpaceError() );
+		}
+
+		const domainParts = `${ this.domain }`.split( '.' );
+
+		if ( value.toLowerCase().includes( domainParts[ 0 ].toLowerCase() ) ) {
+			this.addError( field, PasswordValidator.getPasswordContainsDomainError( domainParts[ 0 ] ) );
+		}
+
+		if ( this.mailboxName && value.toLowerCase().includes( this.mailboxName.toLowerCase() ) ) {
+			this.addError(
+				field,
+				PasswordValidator.getPasswordContainsMailboxNameError( this.mailboxName )
+			);
 		}
+
+		try {
+			const apiResponse = await this.mockApiCall( value );
+			if ( ! apiResponse.success && apiResponse.errors ) {
+				apiResponse.errors.forEach( ( error ) => this.addError( field, error ) );
```

Comment:
Something like this might be fine if we're displaying the actual errors coming from the back end. If we want to customize or translate those errors on the front end, we'll need a `getApiValidationErrors` method.